web/src/settings_preferences.ts

```diff
@@ -27,7 +27,10 @@ type SettingsPanel = {
     notification_sound_elem: string | null;
 };
 
-type UserSettingsProperty = keyof UserSettings;
+type UserSettingsProperty = Exclude<
+    keyof UserSettings,
+    "available_notification_sounds" | "emojiset_choices"
+>;
```

Comment:
Otherwise `let value = user_settings[property]` in `settings_preferences.update_page` can be an array and `settings_components.set_input_element_value($input_elem, value)` is ill-typed. This should be okay because `available_notification_sounds` and `emojiset_choices` can’t actually change or generate events (but why are they in the settings object?).